### PR TITLE
NodeJS - Week1

### PR DESCRIPTION
**Pull request by maribelpreite**

### CODE REVIEW

File: api/src/index.js [hunk 1]
```diff
@@ -3,28 +3,92 @@ import express from "express";
 import cors from "cors";
 import bodyParser from "body-parser";
 import knex from "./database_client.js";
-import nestedRouter from "./routers/nested.js";
+//import mealsRouter from "./routers/meals.js"; --cannot use it yet, but tried
```
Comment (babak-f) [inline api/src/index.js]: Let's get into the habit of not leaving commented out code in the commit :)

File: api/src/index.js [hunk 2]
```diff
@@ -3,28 +3,92 @@ import express from "express";
...
+}); */
+
+
+apiRouter.get("/meals", async (req, res) => {
+  try { //added try/catch to handle errors & removed ternary operator from example as db client is always mysql
+    const SHOW_ALL_MEALS_QUERY = "SELECT * FROM meals ORDER BY `when` desc;";
+    const [meals] = await knex.raw(SHOW_ALL_MEALS_QUERY); //meals in [] to avoid seeing metadata
+    res.json(meals);
+  } catch (error) {
+    res.status(500).json({error: "An error ocurred while trying to fetch data."})
```
Comment (babak-f) [inline api/src/index.js]: Good!

File: api/src/index.js [hunk 3]
```diff
@@ -3,28 +3,92 @@ import express from "express";
...
+    } else {
+      res.json(firstMeal[0]);
+    }
+
+  } catch (error) {
+    res.status(500).json({error: "An error ocurred while trying to fetch the first meal."})
+  }
+})
+
+apiRouter.get("/meals/last", async (req, res) => {
```
Comment (babak-f) [inline api/src/index.js]: Since all your routes start with /meals, you could move /meals to the app.use call below, to avoid repeating it every time :)

**Review by babak-f — APPROVED** [review]
Good job

**Review by magdazelena — APPROVED** [review]
looking good! I agree with Babak's comments for the things to fix up! Also maybe try looking into knex functions instead of using raw statements? It seems like you have quite a good grasp of the material to try new things :)

**Pull request merged**